File: gyroTest/src/main/java/com/stulnikov/arduino/arduinoBt/gyrotest/gyrotest/BasePresenter.java
package com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest;

import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.bluetooth.BlueToothManager;
import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.sensor.AccelerometerProvider;

/**
 * @author alexeystulnikov 12/25/16.
 */

public abstract class BasePresenter<T extends BaseView> {
    private T mView;

    public void bindView(T view) {
        mView = view;
    }

    public void unbindView() {
        mView = null;
    }

    public T getView() {
        return mView;
    }

    public abstract void start();

    public abstract void stop();

    public abstract void setSensorController(AccelerometerProvider provider);

    public abstract void setBluetoothManager(BlueToothManager manager);
}

File: gyroTest/src/main/kotlin/com/stulnikov/arduino/arduinoBt/gyrotest/gyrotest/BasePresenter.kt
package com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest

import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.bluetooth.BluetoothController
import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.sensor.AccelerometerProvider

/**
 * @author alexeystulnikov 12/25/16.
 */
abstract class BasePresenter<T : BaseView?> {
    var view: T? = null
        private set

    fun bindView(view: T) {
        this.view = view
    }

    fun unbindView() {
        view = null
    }

    abstract fun start()
    abstract fun stop()
    abstract fun setSensorController(provider: AccelerometerProvider?)
    abstract fun setBluetoothManager(controller: BluetoothController)
}